src/ds/openapi.h

```diff
@@ -240,6 +240,19 @@ namespace ds
       }
     }
 
+    // This adds a schema description of T to the object j, potentially
+    // modifying another part of the given Doc (for instance, by adding the
+    // schema to a shared component in the document, and making j be a reference
+    // to that). This default implementation simply falls back to
+    // fill_json_schema, which already exists to describe leaf types. A
+    // recursive implementation for struct-to-object types is created by the
+    // json.h macros, and this could be implemented manually for other types.
+    template <typename Doc, typename T>
+    void add_schema_components(Doc&, nlohmann::json& j, const T& t)
+    {
+      fill_json_schema(j, t);
+    }
+
```

Comment:
This lets us remove the `add_schema_components` for `EntityId`, and removes one class of horrid template error when trying to auto-document C++ types - you only need to implement `fill_json_schema`, not this very-similar-but-subtley-different function too.

Reply:
Oh and the side-benefit that I hadn't quite twigged is that it results in more `#ref`s and fewer duplicated schema in the resulting OpenAPI doc.